wallet-new/test/unit/Wallet/Inductive/Cardano.hs

```diff
@@ -248,40 +247,32 @@ equivalentT activeWallet esk = \mkWallet w ->
     toCardano InductiveCtxt{..} fld a = do
         ma' <- catchTranslateErrors $ runIntT' inductiveCtxtInt $ int a
         case ma' of
-          Left err -> throwError EquivalenceNotChecked {
-              equivalenceNotCheckedName   = fld
-            , equivalenceNotCheckedReason = err
-            , equivalenceNotCheckedEvents = inductiveCtxtEvents
-            }
-          Right (a', _ic') ->
-            return a'
+          Left err -> throwError
+              $ EquivalenceNotChecked fld err inductiveCtxtEvents
+          Right (a', _ic') -> return a'
 
 data EquivalenceViolation h =
     -- | Cardano wallet and pure wallet are not equivalent
-    EquivalenceViolation {
-        -- | The property we were checking
-        equivalenceViolationName     :: Text
-
-        -- | Evidence (what was not the same?)
-      , equivalenceViolationEvidence :: EquivalenceViolationEvidence
-
-        -- | The events that led to the error
-      , equivalenceViolationEvents   :: OldestFirst [] (WalletEvent h Addr)
-      }
+    -- EquivalenceViolation
+    --     equivalenceViolationName     = The property we were checking
+    --     equivalenceViolationEvidence = Evidence (what wasn't the same?)
+    --     equivalenceViolationEvents   = The events that led to the error
+    EquivalenceViolation
+        !Text
+        !EquivalenceViolationEvidence
+        !(OldestFirst [] (WalletEvent h Addr))
 
     -- | We got an unexpected interpretation exception
     --
     -- This indicates a bug in the tesing infrastructure.
-  | EquivalenceNotChecked {
-        -- | The property we were checking
-        equivalenceNotCheckedName   :: Text
-
-        -- | Why did we not check the equivalence
-      , equivalenceNotCheckedReason :: IntException
-
-        -- | The events that led to the error
-      , equivalenceNotCheckedEvents :: OldestFirst [] (WalletEvent h Addr)
-      }
+    -- EquivalenceNotChecked
```

Comment:
We should probably use Haddock annotations here to document this constructor along with its fields.

Reply:
I'm unsure if the [new Haddock annotations](https://github.com/haskell/haddock/blob/ghc-8.6/doc/markup.rst#constructors-and-record-fields) for documenting constructor fields work in this project, so I'm going to approve and, if anything, come back later and update the annotations.